app/workers/pull_request_monitor_handlers/path_based_labeler.rb

```diff
@@ -0,0 +1,27 @@
+class PullRequestMonitorHandlers::PathBasedLabeler
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot
+
+  include BranchWorkerMixin
+
+  def perform(branch_id)
+    return unless find_branch(branch_id, :pr)
+    return unless verify_branch_enabled
+
+    process_branch
+  end
+
+  private
+
+  def process_branch
+    label_rules.each do |rule|
+      if diff_file_names.any? { |file_name| file_name =~ Regexp.new(rule.pattern) }
+        GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, [rule.label])
+      end
+    end
+  end
+
+  def label_rules
+    Settings.path_based_labeler.enabled_repos.public_send(fq_repo_name)
```

Comment:
Prefer square bracket access over public_send for dynamic stuff (I'm not sure why enabled_for? was ever written that way, but it's not the preferred way)

```ruby
Settings.path_based_labeler.enabled_repos[fq_repo_name]
```